### DISCUSSION TITLE
Hit Step?

### DISCUSSION THREAD

**Discussion by NerdyViking** [AngryBeaver/beavers-crafting#59] [Q&A] [answered]
Hi!

Is there a way to set up progressive successes? Maybe it has to do with the hit step Test that I can see. Basically like if they roll a 10 they get one result, if they roll a 15 they get another result (maybe two potions or a +1 version of that item).

I understand this is perhaps a silly and complicated desire for crafting, but thought it might be possible perhaps using macros or whatever that hit step function is.

Thanks for any help!

**Comment by AngryBeaver** [comment, marked as answer]
The hit step is just for manual progression that can not be tracked automatically or is too complicated to get tracked automtically. like "kill oger" 10 times you hit that when ever you kill an oger. (This module is no longer only for Crafting Recipes can also be Quests)

I also have played with the idea of adding a "macro type" for tests so a macro will get executed when ever you advance. But this is currently not implemented and will not come any time soon. I first of would like to cleanUp the current macro usage.

For a standard implementation this seems to me too specific and too complicated in both ui simplicity and code.
e.g. you want it to be 2 steps success and greater succes while the next one wants 3 or 4 or something bad that happens on a natural 1 etc.

there is no real alternative currently you might trust your players and add a or condition "hit step" to a "skill check" that way they might with your permission advance for them self when they roll exceptional good.

else you have to wait till i got some time for advanced macros.

**Reply by AngryBeaver** [reply]
this is the awnser for the hit step question.

**Comment by NerdyViking** [comment]
Understood. Thanks for the reply!

Would it be possible to have the roll performed for the check replace the roll formula in the linked table? For example, using a macro

> const table = game.tables.getName("Potion Crafting");
> 
> const roll = new Roll("1d20 + @abilities.dex.mod + @attributes.prof", actor.getRollData());
> 
> table.draw({roll});

Where this just calls for the actor data and updates the roll formula field of the called table before rolling. So, in the example of the crafting system, the macro would store the result of the roll used for the DC check. If it meets the DC, it counts as a success and then rolls on the attached table as per normal, but with the updated roll formula being the stored variable from the initial check.

**Reply by AngryBeaver** [reply]
Not sure i understand you.

How macros currently work: "They are executed at the beginning before crafting." At that moment they can manipulate the recipe you are about to craft, meaning they would be able to add or replace everything costs, ingredients, requires including test checks.

Else than that you might be able to store data on the actor with this macro but that wont effect anything as the recipe would then ofc not do anything with that stored data as there is no other macro execution time. And the recipe ofc does not know about anything you might or might not have stored anywhere else.

Not sure what this has to do with the roll table formula but as said you also might be able to manipulate the recipe so you can have replaced any result you want with what ever else you wanted. Roll Table results are currently rolled at the start of a crafting process and the crafting process anyway has only a fixed result entity. (However there is a feature request that this change).

That said you should be carefull when using powerfull macros currently as they are subjected to break easily between my releases.

Currently macros are extremly powerfull but this also means a lot of what you do will use recipe methods and when ever i am working on this module those methods can change and with it macros that depend on it. E.g. currently i am implementing "or conditions" for ingredients. the recipe will change from a list of costs to a "list of list of costs" -> this ofc can break existing macros that manipulated the ingredients list. 

So at the moment i am a bit unhappy about my macro system ->  so this might change at somepoint "advanced macros" to be more simpler but also more robust and with more possible execution times. (at the start) (at the end) and as (a test check).
